Reed/DefinitionView/viewmodels/DefinitionViewModel.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  DefinitionModalViewModel.swift
+//  Reed
+//
+//  Created by Hugo Zhan on 11/3/20.
+//  Copyright © 2020 Roger Luo. All rights reserved.
+//
+
+import SwiftUI
+
+class DefinitionViewModel: ObservableObject {
+    @Published var title = ""
+    @Published var currReading = ""
+    @Published var termsArr = [Terms]()
+    @Published var defsArr = [Defs]()
+    
+    let dictionaryFetcher = DictionaryFetcher()
+    
+    func generateContent(from tappedWord: String) {
+        title = tappedWord
+        termsArr.removeAll()
+        defsArr.removeAll()
+        let entries = dictionaryFetcher.fetchEntries(of: tappedWord)
+        if entries.count > 0 && entries[0].readings[0].terms.count > 0 && entries[0].readings[0].terms[0] == tappedWord {
+            currReading = entries[0].readings[0].reading
+        } else {
+            currReading = ""
+        }
+        for entry in entries {
+            for read in entry.readings {
+                let reading = read.reading
+                for term in read.terms {
+                    termsArr.append(Terms(term: term, reading: reading))
```

Comment:
Not 100% sure, but I think _each_ call to `append` causes the view to rerender because `termsArr` is `@Published`. It may be better to build this array out first, then assign the finalized result to `termsArr`.